src/synapse/azext_synapse/custom.py

```diff
@@ -131,17 +134,40 @@ def update_workspace(cmd, client, resource_group_name, workspace_name, sql_admin
     return sdk_no_wait(no_wait, client.update, resource_group_name, workspace_name, workspace_patch_info)
 
 
-def get_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name):
+def custom_check_name_availability(cmd, client, name):
+    return client.check_name_availability(name, "Microsoft.Synapse/workspaces")
+
+
+def get_spark_pool(cmd, client, workspace, spark_pool_name):
+    parsed_workspace = extract_workspace_and_resource_group(workspace)
+    workspace_name = parsed_workspace['name']
+    resource_group_name = parsed_workspace['resource_group']
     return client.get(resource_group_name, workspace_name, spark_pool_name)

```

Comment:
> This method looks sort of strange to me. It requires users to pass workspace resource id/name and spark pool name. I would expect we support 1. workspace name and spark pool name 2. spark pool resource id.

For this, it supports both 1 and 2. It also supports workpace id and spark pool name

Reply:
Correct, that is what I am thinking. We can discuss offline and reference child resource scenarios of existing services.

Reply:
> Correct, that is what I am thinking. We can discuss offline and reference child resource scenarios of existing services.

There are another two points we need to concern:
1. The parameter name is workspace not workspace-name. As you know liking storage account, cli supports to use storage-account to support both id and name instead of storage-account-name and storage-account-id
2. We need resource group as a parameter. How can we get the workspace's resource group?
(a) calling the get workspace api (b) getting the workspace's id and parsed the id

So this is why we need lines 141,142 and 143.

Reply:
> This method looks sort of strange to me. It requires users to pass workspace resource id/name and spark pool name. I would expect we support 1. workspace name and spark pool name 2. spark pool resource id.

Fixed